### PR TITLE
[API] Soft labelling losses refactor

### PR DESCRIPTION
**Pull request by victormvy**
I have reimplemented the classes used for the soft labelling losses to allow their usage with an arbitrary loss function, not only the CrossEntropyLoss. In this way, several classes have been changed and renamed:
- `BetaCrossEntropyLoss` now is `BetaLoss`.
- `BinomialCrossEntropyLoss` now is `BinomialLoss`.
- `CustomTargetsCrossEntropyLoss` now is `CustomTargetsLoss`.
- `ExponentialCrossEntropyLoss` now is `ExponentialLoss`.
- `GeneralTriangularCrossEntropyLoss` now is `GeneralTriangularLoss`.
- `PoissonCrossEntropyLoss` now is `PoissonLoss`.
- `TriangularCrossEntropyLoss` now is `TriangularLoss`.

They all have a `base_loss` parameter that can be used to pass the base loss function object that will be used. Thus, the new classes are loss function wrappers that transform the hard labels to soft labels.

### CODE REVIEW

File: tutorials/dlordinal_with_skorch_tutorial.ipynb
Comment (dguijo) [inline tutorials/dlordinal_with_skorch_tutorial.ipynb:1]: Reinicia el contador de celdas.

File: tutorials/hybrid_dropout_tutorial.ipynb
Comment (dguijo) [inline tutorials/hybrid_dropout_tutorial.ipynb:1]: aquí igual

**Review by dguijo — APPROVED** [review]
LGTM!

**Review by pagutierrez — APPROVED** [review]
Great, much better and generic now
